app/controllers/api/current/blood_sugars_controller.rb

```diff
@@ -1,15 +1,49 @@
 class Api::Current::BloodSugarsController < Api::Current::SyncController
+  include Api::Current::PrioritisableByFacility
+  include Api::Current::SyncEncounterObservation
 
   def sync_from_user
-    render json: { errors: [] }, status: :ok
+    __sync_from_user__(blood_sugars_params)
   end
 
   def sync_to_user
-    render json: {
-      blood_sugars: [],
-      process_token: ""
-    }, status: :ok
+    __sync_to_user__('blood_sugars')
 
   end
 
+  private
+
+  def transform_to_response(blood_sugar)
+    Api::Current::Transformer.to_response(blood_sugar)
+  end
+
+  def merge_if_valid(blood_sugar_params)
+    validator = Api::Current::BloodSugarPayloadValidator.new(blood_sugar_params)
+    logger.debug "Blood Sugar payload had errors: #{validator.errors_hash}" if validator.invalid?
+    if validator.invalid?
+      NewRelic::Agent.increment_metric('Merge/BloodSugar/schema_invalid')
+      { errors_hash: validator.errors_hash }
+    else
+      set_patient_recorded_at(blood_sugar_params)
```

Comment:
I think we don't need this for `BloodSugars` since this is a fresh API, so we need not set the `recorded_at` in real-time as all new apps who have support for Diabetes will send `recorded_at`

Reply:
We still need to set_patient_recorded_at for retroactive BloodSugar records